src/Components/Layout.tsx

```diff
@@ -1,596 +1,344 @@
-          <TabPanel>
-            <MenuView {...getMenuObj()} />
-          </TabPanel>
-          <TabPanel>
-            <Body {...getBodyObj(0)} {...getDateSelectObj()} />
-          </TabPanel>
-          <TabPanel>
-            <Body {...getBodyObj(1)} {...getDateSelectObj()} />
-          </TabPanel>
-        </Tabs>
-      </div>
-    </MuiThemeProvider>
-  );
-};
+import React, { useState, useContext } from "react";
+import { Button, Box } from "@material-ui/core";
+import { DateRange } from "./DateRange";
+import { Tab, Tabs, TabList, TabPanel } from "react-tabs";
+import { Body } from "./Body";
+import { MenuView } from "./Menu";
+import { TimerUI } from "./Timer";
+import { QuickSelect } from "./QuickSelect";
+import CalendarTodayIcon from "@material-ui/icons/CalendarToday";
+import KeyboardTabIcon from "@material-ui/icons/KeyboardTab";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+import RefreshIcon from "@material-ui/icons/Refresh";
+import "./Styling.css";
+import { MuiThemeProvider } from "@material-ui/core/styles";
+import { GlobalContext } from "./Constants";
+import {
+  DateRangeUI,
+  uiData,
+  dropdownData as dData,
+  DropdownData,
+  refreshData as rData,
+  RefreshData,
+  bodyConfig as bConfig,
+  BodyConfig,
+} from "./Types";
+import { style as themeStyle } from "./Style";
+
+interface Inputs {
+  onTimerEvent(): void;
+  onDateEvent(dates: Date[]): void;
+  dateFormatter?: Intl.DateTimeFormat;
+  theme?: any;
+  commonlyUsedText?: number[];
+  quickSelectTerms?: string[];
+  quickSelectIntervals?: number[];
+  relativeTerms?: string[];
+  relativeIntervals?: number[];
+  localeString?: string;
+  nowText?: string;
+  minimumYearValue?: number;
+  maximumYearValue?: number;
+  humanizer?: any;
+  setStoredRange?(dateRange: string[]): void;
+  storedRange: string[] | null;
+}
+
+export const Layout: React.FC<Inputs> = (props) => {
+  // GLOBAL
+  let defaults = useContext(GlobalContext);
+  defaults.classes = themeStyle();
+  defaults.commonlyUsedText =
+    props.commonlyUsedText || defaults.commonlyUsedText;
+
+  defaults.quickSelectIntervals =
+    props.quickSelectIntervals || defaults.quickSelectIntervals;
+
+  defaults.relativeIntervals =
+    props.relativeIntervals || defaults.quickSelectIntervals;
+
+  defaults.minimumYearValue =
+    props.minimumYearValue || defaults.minimumYearValue;
+
+  defaults.maximumYearValue =
+    props.maximumYearValue || defaults.maximumYearValue;
+
+  defaults.localeObj = {
+    localeString: props.localeString || defaults.localeObj.localeString,
+    dateFormatter: props.dateFormatter || defaults.localeObj.dateFormatter,
+    quickSelectTerms:
+      props.quickSelectTerms || defaults.localeObj.quickSelectTerms,
+    relativeTerms: props.relativeTerms || defaults.localeObj.relativeTerms,
+    nowText: props.nowText || defaults.localeObj.nowText,
+    humanizer: props.humanizer || defaults.localeObj.humanizer,
+  };
+
+  // STORAGE
+  const setStoredRange:
```

Comment:
Why doesn't this live in the DateRange object?

Reply:
Done

Reply:
Done